kvbc/test/v4blockchain/v4_blockchain_test.cpp

```diff
@@ -740,6 +843,770 @@ TEST_F(v4_kvbc, delete_last_reachable) {
+  for (size_t i = 0; i < versioned_keys.size(); ++i) {
+    if (deleted_keys.count(versioned_keys[i])) {
+      ASSERT_FALSE(values[i].has_value());
+    } else {
+      ASSERT_TRUE(values[i].has_value());
+      auto versioned_val = std::get<categorization::VersionedValue>(*(values[i]));
+      ASSERT_EQ(versioned_keys[i].substr(key_start), (versioned_val.data).substr(val_start));
+    }
+  }
+
+  blockchain.multiGet("immutable", immutable_keys, immutable_versions, values);
+  key_start = std::string("immutable_key_").size();
+  val_start = std::string("immutable_value_").size();
+  for (size_t i = 0; i < immutable_keys.size(); ++i) {
+    if (deleted_keys.count(immutable_keys[i])) {
+      ASSERT_FALSE(values[i].has_value());
+    } else {
+      ASSERT_TRUE(values[i].has_value());
+      auto immutable_val = std::get<categorization::ImmutableValue>(*(values[i]));
+      ASSERT_EQ(immutable_keys[i].substr(key_start), (immutable_val.data).substr(val_start));
+    }
+  }
+}
+
+TEST_F(v4_kvbc, all_get_latest) {
+  uint64_t max_block = 100;
+  uint32_t num_merkle_each = 0;
+  uint32_t num_versioned_each = 0;
+  uint32_t num_immutable_each = 0;
+  create_blocks(max_block, num_merkle_each, num_versioned_each, num_immutable_each);
+
+  std::map<std::string, categorization::CATEGORY_TYPE> cat_map{
+      {"merkle", categorization::CATEGORY_TYPE::block_merkle},
+      {"versioned", categorization::CATEGORY_TYPE::versioned_kv},
+      {"immutable", categorization::CATEGORY_TYPE::immutable}};
+  v4blockchain::KeyValueBlockchain blockchain{db, true, cat_map};
+  ASSERT_EQ(blockchain.getGenesisBlockId(), 1);
+  ASSERT_EQ(blockchain.getLastBlockId(), max_block);
+
+  std::vector<std::string> merkle_keys;
+  std::vector<std::string> versioned_keys;
+  std::vector<std::string> immutable_keys;
+
+  ///////Checking the get///////////////////////
+  for (uint64_t blk = 1; blk <= max_block; ++blk) {
+    for (uint32_t kid = 1; kid <= num_merkle_each; ++kid) {
+      std::string key = "merkle_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+      std::string val_str = "merkle_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+      merkle_keys.push_back(key);
+      auto val = blockchain.getLatest("merkle", key);
+      ASSERT_TRUE(val.has_value());
+      auto merkle_val = std::get<categorization::MerkleValue>(*val);
+      ASSERT_EQ(merkle_val.block_id, max_block);
+      ASSERT_EQ(merkle_val.data, val_str);
+    }
+
+    for (uint32_t kid = 1; kid <= num_versioned_each; ++kid) {
+      std::string key = "versioned_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+      std::string val_str = "versioned_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+      versioned_keys.push_back(key);
+      auto val = blockchain.getLatest("versioned", key);
+      ASSERT_TRUE(val.has_value());
+      auto ver_val = std::get<categorization::VersionedValue>(*val);
+      ASSERT_EQ(ver_val.block_id, max_block);
```

Comment:
this will need to be changed to the real block id that the key was added

Reply:
Thanks, @cloudnoize for this comment. This makes everything symmetric now.